cirq/google/op_serializer.py

```diff
@@ -66,31 +66,44 @@ class GateOpSerializer:
         serialized_gate_id: The id used when serializing the gate.
     """
 
-    def __init__(self, *, gate_type: Type[Gate], serialized_gate_id: str,
-                 args: List[SerializingArg]):
+    def __init__(self,
+                 *,
+                 gate_type: Type[Gate],
+                 serialized_gate_id: str,
+                 args: List[SerializingArg],
+                 gate_predicate: Callable[[ops.Gate], bool] = None):
         """Construct the serializer.
 
         Args:
             gate_type: The type of the gate that is being serialized.
             serialized_gate_id: The string id of the gate when serialized.
+            gate_predicate: Sometimes a gate can only be serialized for
```

Comment:
Maybe a more descriptive name than `gate_predicate`? `can_serialize_predicate`? This is a `GateOpSerializer` so presumably everything has to do with gates

Reply:
done